pkg/pattern/drain/chunk.go

```diff
@@ -21,16 +24,20 @@ const (
 type Chunks []Chunk
 
 type Chunk struct {
-	Samples []logproto.PatternSample
+	Samples            []logproto.PatternSample
+	StructuredMetadata map[string]string
 }
 
-func newChunk(ts model.Time) Chunk {
+func newChunk(ts model.Time, structuredMetadata push.LabelsAdapter) Chunk {
```

Comment:
I'm not confident I am using the right types to pass the labels through from the request - any recommendations here?